### ISSUE TITLE
improve Steam cache cleaning rules

### ISSUE THREAD

**Issue by FuccDucc** [MoscaDotTo/Winapp2#817]
Hi,

I found some Steam for Windows ([https://store.steampowered.com/about/](https://store.steampowered.com/about/)) cache locations that aren't included in Winapp2's current set of cache cleaning rules for this application.

See the improvement suggestions below:

To also include..

**General**
C:\Program Files (x86)\Steam\appcache\librarycache
C:\Program Files (x86)\Steam\userdata\[id-here]\inventorymsgcache
C:\Program Files (x86)\Steam\userdata\[id-here]\config\librarycache
C:\Program Files (x86)\Steam\steam\cached

**Steam Crash Reports**
C:\Program Files (x86)\Steam\dumps\reports > any contents (usually contains .dmp crash dumps)

**Steam Download Cache**
C:\Program Files (x86)\Steam\steamapps\downloading

**Steam Game Skin Cache**
C:\Program Files (x86)\Steam\steamapps\workshop\content\[unique appid]\ > any folder, they are numbered asset id cache folders that contain local textures
C:\Program Files (x86)\Steam\steamapps\workshop\temp > any contents
C:\Program Files (x86)\Steam\steamapps\workshop\downloads > any contents

**Fix some logs cleaning**
C:\Program Files (x86)\Steam\logs > any folder content
* For some reason, not all log .txt files are being cleared by existing Winapp2 rule for Steam logs.. only some. So better include the folder in a separate rule, and entirely.

**Particular cache files:**
C:\Program Files (x86)\Steam\userdata\[local profile SteamID3 here]\[unique appid] > remotecache.vdf
* Realistic example: C:\Program Files (x86)\Steam\userdata\91324041\730\remotecache.vdf

C:\Program Files (x86)\Steam\userdata\[local profile SteamID3 here]\[unique appid]\local > socache.dt
* Realistic example: C:\Program Files (x86)\Steam\userdata\91324041\730\local\socache.dt

**Comment by Shakil-Shahadat** [comment]
See if this works.
```
[Steam for Windows *]
LangSecRef=3021
DetectFile=%ProgramFiles%\Steam
FileKey1=%ProgramFiles%\Steam\appcache\librarycache|*.*|RECURSE
FileKey2=%ProgramFiles%\Steam\dumps\reports|*.*|RECURSE
FileKey3=%ProgramFiles%\Steam\logs|*.*|RECURSE
FileKey4=%ProgramFiles%\Steam\steam\cached|*.*|RECURSE
FileKey5=%ProgramFiles%\Steam\steamapps\downloading|*.*|RECURSE
FileKey6=%ProgramFiles%\Steam\steamapps\workshop\content*|*.*|RECURSE
FileKey7=%ProgramFiles%\Steam\steamapps\workshop\downloads|*.*|RECURSE
FileKey8=%ProgramFiles%\Steam\steamapps\workshop\temp|*.*|RECURSE
FileKey9=%ProgramFiles%\Steam\userdata\*\*|remotecache.vdf
FileKey10=%ProgramFiles%\Steam\userdata\*\*\local|socache.dt
FileKey11=%ProgramFiles%\Steam\userdata*\config\librarycache|*.*|RECURSE
FileKey12=%ProgramFiles%\Steam\userdata*\inventorymsgcache|*.*|RECURSE
```
Little confused about these two though.
```
C:\Program Files (x86)\Steam\userdata[id-here]\inventorymsgcache
C:\Program Files (x86)\Steam\userdata[id-here]\config\librarycache
```
Are there any '\\' immediately after 'userdata'? A (semi) realistic example might help.

**Comment by FuccDucc** [comment]
> Are there any '\' immediately after 'userdata'?

yeah i made a typo.. it is:
```
C:\Program Files (x86)\Steam\userdata\[id-here]\inventorymsgcache
C:\Program Files (x86)\Steam\userdata\[id-here]\config\librarycache
``` 
and ill fix it in my main post as well. So examples would be:

C:\Program Files (x86)\Steam\userdata\91324041\inventorymsgcache
C:\Program Files (x86)\Steam\userdata\91324041\config\librarycache

> One more thing, is 'Steam for Windows' already supported in CCleaner or in Winapp2? If so, how is it named? 'Steam for Windows' or just 'Steam'?

it is [supported by Winapp2](https://github.com/MoscaDotTo/Winapp2/blob/master/Winapp2.ini#L19780) only, but is missing some caches. That's why i made this issue. So you probably gotta change [Steam for Windows *] to [Steam *] ...

Actually i noticed the separation between categories (in the existing Winapp2 steam entries) so you'd have to split the updates into each appropiate one as well. To keep in mind all suggestions would have to be added, even while some seem like duplicates (unless you can find out why those don't delete everything they aim to, and fix them 1 by 1.. like the Logs folder content)

**Comment by APMichael** [comment]
> ... 
> One more thing, is 'Steam for Windows' already supported in CCleaner or in Winapp2? If so, how is it named? 'Steam for Windows' or just 'Steam'?

Correct, Steam is already supported by CCleaner.

The following two FileKeys are not needed because they are already included in the built-in CCleaner entry.

```
FileKey2=%ProgramFiles%\Steam\dumps\reports|*.*|RECURSE
FileKey3=%ProgramFiles%\Steam\logs|*.*|RECURSE
```

CCleaner [Steam]:
```
FileKey2=%ProgramFiles%\Steam\Dumps|*.*|RECURSE
FileKey3=%ProgramFiles%\Steam\Logs|*.*|RECURSE
```

The other FileKeys would have to be sorted into the following Winapp2.ini entries according to their affiliation:
```
[Steam *]
[Steam Cache *]
[Steam Games *]
```

**Comment by APMichael** [comment]
> yeah, looks like a typo (GitHub editor has a bug.. it displays `\` in "Edit" but not in the post). ...

JFYI: GitHub uses "Styling with Markdown" in the comments. Therefore, you have to escape some special characters with a backslash "\\". The special characters are for example the asterisk "\*" and the backslash "\\" itself (escaping: "\\\*" or "\\\\").

**Comment by Shakil-Shahadat** [comment]
See if this classification is correct. Ignore the lack of numbering.
```
[Steam *]

[Steam Cache *]
FileKey=%ProgramFiles%\Steam\appcache\librarycache|*.*|RECURSE

FileKey=%ProgramFiles%\Steam\steam\cached|*.*|RECURSE

FileKey=%ProgramFiles%\Steam\userdata\*\*|remotecache.vdf
FileKey=%ProgramFiles%\Steam\userdata\*\*\local|socache.dt
FileKey=%ProgramFiles%\Steam\userdata\*\config\librarycache|*.*|RECURSE
FileKey=%ProgramFiles%\Steam\userdata\*\inventorymsgcache|*.*|RECURSE

[Steam Games *]
FileKey=%ProgramFiles%\Steam\steamapps\downloading|*.*|RECURSE
FileKey=%ProgramFiles%\Steam\steamapps\workshop\content*|*.*|RECURSE
FileKey=%ProgramFiles%\Steam\steamapps\workshop\downloads|*.*|RECURSE
FileKey=%ProgramFiles%\Steam\steamapps\workshop\temp|*.*|RECURSE
```
Here I fixed,
- `userdata\[id-here]` problems
- removed two redundant keys.
- title

Also, the backtick works well for escaping. And triple backtick at the start and at the end of a code block.

Another question, how do I know what CCleaner supports or not? Is there a winapp1.ini file?

**Comment by FuccDucc** [comment]
> C:\Program Files (x86)\Steam\steamapps\workshop\content[unique appid]\ > any folder, they are numbered asset id cache folders that contain local textures

Maybe add a warning dialog for this entry stating: "After using this, it's recommended to run Steam "Verify game files" for all games".. clearing this means that game skins may load sluggishly afterwards, and a verify restores them to exactly what the game needs on the spot.

**Comment by MoscaDotTo** [comment]
remotecache.vdf may have to do with steam cloud syncing

**Comment by Shakil-Shahadat** [comment]
A warning can be added per-app basis not per FileKey basis if I'm not mistaken.

**Comment by FuccDucc** [comment]
> remotecache.vdf may have to do with steam cloud syncing

yes, but it's a cache and if cleared, has no effect on loading of the actual cloud or local save files

- 'remote' folder: it contains all the files that are being synced up with Steam servers
- remotecache.vdf file (directly below 'remote' folder): caches the properties (such as file size or modification date) of the actual cloud save files contained in 'remote' folder

The remotecache.vdf file is plaintext once edited in Notepad.. example contents:
```
"730"
{
	"ChangeNumber"		"0"
	"cfg/csgo_saved_item_shuffles.txt"
	{
		"root"		"0"
		"size"		"24"
		"localtime"		"1522861973"
		"time"		"1522861973"
		"remotetime"		"1522861973"
		"sha"		"c7bdc563982fc4ddddb9e6f8853b298967d858f9"
		"syncstate"		"1"
		"persiststate"		"0"
		"platformstosync2"		"-1"
	}
}
```

**Comment by CSGalloway** [comment]
You can get documentation on creating entries:
https://github.com/MoscaDotTo/Winapp2/#creating-entries

Also if you:
CCleaner.exe /EXPORT .

it will export:
winapp.ini
winreg.ini
winsys.ini

http://www.facebook.com/CSGalloway

**Comment by APMichael** [comment]
> See if this classification is correct. Ignore the lack of numbering.

Thank you very much for your help.

I don't use Steam and can't really help with that, but the classification looks good to me.

Possibly a separate entry (e.g. `[Steam User Data Cache *]`) with a warning should be created for the FileKeys with "userdata" in the path?

> Another question, how do I know what CCleaner supports or not? Is there a winapp1.ini file?

As CSGalloway has already written: Use the command `CCleaner.exe /export` and you will find the files "winapp.ini" and "winsys.ini" in the CCleaner folder.

**Comment by Shakil-Shahadat** [comment]
So, here is the updated entries. I have added a # at the end of the lines where new entries are added to identify them clearly.
```
[Steam Cache *]
Section=Games
Detect=HKCU\Software\Valve\Steam
FileKey1=%LocalAppData%\Steam\htmlcache|*.*|RECURSE
FileKey2=%ProgramFiles%\Steam\appcache\httpcache|*.*|RECURSE
FileKey3=%ProgramFiles%\Steam\appcache\librarycache|*.*|RECURSE	#
FileKey4=%ProgramFiles%\Steam\config\cookies|*.*|RECURSE
FileKey5=%ProgramFiles%\Steam\config\htmlcache|*.*|RECURSE
FileKey6=%ProgramFiles%\Steam\config\overlay*|*.*|RECURSE
FileKey7=%ProgramFiles%\Steam\depotcache|*.*
FileKey8=%ProgramFiles%\Steam\steam\cached|*.*|RECURSE	#
FileKey9=%ProgramFiles%\Steam\steamapps\shadercache|*.*|RECURSE
FileKey10=%ProgramFiles%\Steam\tenfoot\config\httpcache|*.*|RECURSE
FileKey11=%ProgramFiles%\Steam\tenfoot\config\images\web|*.*|RECURSE
FileKey12=%ProgramFiles%\Steam\userdata\*\ugcmsgcache|*.*|RECURSE

[Steam Games *]
Section=Games
Detect=HKCU\Software\Valve\Steam
FileKey1=%ProgramFiles%\Steam\steamapps|*.dmp;.DS_Store;*.icns;logfile.*;*.log;*.mdmp;*.tmp;*log.txt;log*.txt;text.txt|RECURSE
FileKey2=%ProgramFiles%\Steam\steamapps\common\*\*\downloads|*.*|RECURSE
FileKey3=%ProgramFiles%\Steam\steamapps\common\*\Cache|*.*|RECURSE
FileKey4=%ProgramFiles%\Steam\steamapps\common\*\config\html|*.*|RECURSE
FileKey5=%ProgramFiles%\Steam\steamapps\common\*\DebugData|*.*|RECURSE
FileKey6=%ProgramFiles%\Steam\steamapps\downloading|*.*|RECURSE	#
FileKey7=%ProgramFiles%\Steam\Steamapps\temp|*.*|RECURSE
FileKey8=%ProgramFiles%\Steam\steamapps\workshop\content*|*.*|RECURSE	#
FileKey9=%ProgramFiles%\Steam\steamapps\workshop\downloads|*.*|RECURSE	#
FileKey10=%ProgramFiles%\Steam\steamapps\workshop\temp|*.*|RECURSE	#
ExcludeKey1=PATH|%ProgramFiles%\Steam\steamapps\common\Supreme Ruler 1936\Cache\|*.*
ExcludeKey2=PATH|%ProgramFiles%\Steam\steamapps\common\Supreme Ruler Cold War\Cache\|*.*
ExcludeKey3=PATH|%ProgramFiles%\Steam\steamapps\common\Supreme Ruler Ultimate\Cache\|*.*


[Steam User Data Cache *]
Section=Games
Detect=HKCU\Software\Valve\Steam
Warning=
FileKey1=%ProgramFiles%\Steam\userdata\*\*|remotecache.vdf	#
FileKey2=%ProgramFiles%\Steam\userdata\*\*\local|socache.dt	#
FileKey3=%ProgramFiles%\Steam\userdata\*\config\librarycache|*.*|RECURSE	#
FileKey4=%ProgramFiles%\Steam\userdata\*\inventorymsgcache|*.*|RECURSE	#
```
Two things.
- There is another `userdata` entry in the `[Steam Cache *]` section. Should I move that too to `[Steam User Data Cache *]` section?
- What should be the warning message for `[Steam User Data Cache *]`?

Also, tried the export command, it worked.

**Comment by APMichael** [comment]
> * There is another `userdata` entry in the `[Steam Cache *]` section. Should I move that too to `[Steam User Data Cache *]` section?

Yes, for me it would make sense to move the FileKey.

> * What should be the warning message for `[Steam User Data Cache *]`?

@FuccDucc had mentioned the warning in a post:

`Warning=After using this, it's recommended to run Steam's "Verify integrity of game files" for all games.`

**Comment by APMichael** [comment]
Oh, sorry, I got something mixed up there. I guess the warning is only for the FileKeys with "\\workshop\\content\\[unique appid]". Then these FileKeys would need their own entry and not the FileKeys with "userdata".

Since I made the mistake, I briefly corrected the entries for you:


```
[Steam Cache *]
Section=Games
Detect=HKCU\Software\Valve\Steam
FileKey1=%LocalAppData%\Steam\htmlcache|*.*|RECURSE
FileKey2=%ProgramFiles%\Steam\appcache\httpcache|*.*|RECURSE
FileKey3=%ProgramFiles%\Steam\appcache\librarycache|*.*|RECURSE #
FileKey4=%ProgramFiles%\Steam\config\cookies|*.*|RECURSE
FileKey5=%ProgramFiles%\Steam\config\htmlcache|*.*|RECURSE
FileKey6=%ProgramFiles%\Steam\config\overlay*|*.*|RECURSE
FileKey7=%ProgramFiles%\Steam\depotcache|*.*
FileKey8=%ProgramFiles%\Steam\steam\cached|*.*|RECURSE #
FileKey9=%ProgramFiles%\Steam\steamapps\shadercache|*.*|RECURSE
FileKey10=%ProgramFiles%\Steam\tenfoot\config\httpcache|*.*|RECURSE
FileKey11=%ProgramFiles%\Steam\tenfoot\config\images\web|*.*|RECURSE
FileKey12=%ProgramFiles%\Steam\userdata\*\*|remotecache.vdf #
FileKey13=%ProgramFiles%\Steam\userdata\*\*\local|socache.dt #
FileKey14=%ProgramFiles%\Steam\userdata\*\config\librarycache|*.*|RECURSE #
FileKey15=%ProgramFiles%\Steam\userdata\*\inventorymsgcache|*.*|RECURSE #
FileKey16=%ProgramFiles%\Steam\userdata\*\ugcmsgcache|*.*|RECURSE

[Steam Game Skin Cache *]
Section=Games
Detect=HKCU\Software\Valve\Steam
Warning=After using this, it's recommended to run Steam's "Verify integrity of game files" for all games.
FileKey1=%ProgramFiles%\Steam\steamapps\workshop\content\*|*.*|RECURSE #

[Steam Games *]
Section=Games
Detect=HKCU\Software\Valve\Steam
FileKey1=%ProgramFiles%\Steam\steamapps|*.dmp;.DS_Store;*.icns;logfile.*;*.log;*.mdmp;*.tmp;*log.txt;log*.txt;text.txt|RECURSE
FileKey2=%ProgramFiles%\Steam\steamapps\common\*\*\downloads|*.*|RECURSE
FileKey3=%ProgramFiles%\Steam\steamapps\common\*\Cache|*.*|RECURSE
FileKey4=%ProgramFiles%\Steam\steamapps\common\*\config\html|*.*|RECURSE
FileKey5=%ProgramFiles%\Steam\steamapps\common\*\DebugData|*.*|RECURSE
FileKey6=%ProgramFiles%\Steam\steamapps\downloading|*.*|RECURSE #
FileKey7=%ProgramFiles%\Steam\Steamapps\temp|*.*|RECURSE
FileKey8=%ProgramFiles%\Steam\steamapps\workshop\downloads|*.*|RECURSE #
FileKey9=%ProgramFiles%\Steam\steamapps\workshop\temp|*.*|RECURSE #
ExcludeKey1=PATH|%ProgramFiles%\Steam\steamapps\common\Supreme Ruler 1936\Cache\|*.*
ExcludeKey2=PATH|%ProgramFiles%\Steam\steamapps\common\Supreme Ruler Cold War\Cache\|*.*
ExcludeKey3=PATH|%ProgramFiles%\Steam\steamapps\common\Supreme Ruler Ultimate\Cache\|*.*
```


 Sorry again.

**Comment by MoscaDotTo** [comment]
These changes are included in #824
